xds/internal/xdsclient/clientimpl_watchers.go

```diff
@@ -26,6 +26,24 @@ import (
 	"google.golang.org/grpc/xds/internal/xdsclient/xdsresource"
 )
 
+// wrappingWatcher is a wrapper around an xdsresource.ResourceWatcher that adds
+// the node ID to the error messages reported to the watcher.
+type wrappingWatcher struct {
+	nodeID  string
+	watcher xdsresource.ResourceWatcher
+}
+
+func (w *wrappingWatcher) OnUpdate(update xdsresource.ResourceData, done xdsresource.OnDoneFunc) {
+	w.watcher.OnUpdate(update, done)
+}
```

Comment:
Optional: embed and delete the pass-through methods?  (i.e. make the PR even smaller 😄 )

Reply:
Sigh. I promise, that was the first thing I thought about and somehow my brain told me that I can't do it. I blame it on the ongoing Rust learning :P